inc/vcf/parsing_state.hpp

```diff
@@ -41,22 +41,21 @@ namespace ebi
         bool m_is_valid;
         
         std::shared_ptr<Source> source;
-        std::shared_ptr<std::vector<Record>> records;
-        std::shared_ptr<std::vector<std::unique_ptr<Error>>> errors;
-        std::shared_ptr<std::vector<std::unique_ptr<Error>>> warnings;
+        std::unique_ptr<Record> record;
+        std::vector<std::unique_ptr<Error>> errors;
+        std::vector<std::unique_ptr<Error>> warnings;
 
         std::multimap<std::string, std::string> defined_metadata;
         std::multimap<std::string, std::string> undefined_metadata;
         
-        ParsingState(std::shared_ptr<Source> source,
-                     std::shared_ptr<std::vector<Record>> records);
+        ParsingState(std::shared_ptr<Source> source);
         
-        void set_version(Version version) const;
+        void set_version(Version version);
         
-        void add_meta(MetaEntry const & meta) const;
-        
-        void add_record(Record const & record);
-        void clear_records() const;
+        void add_meta(MetaEntry const & meta);
+
+        void add_record(std::unique_ptr<Record> record);
+        void clear_records();
```

Comment:
Rename the first to `set_record` and maybe remove the second one? Otherwise it could be named `unset_record` instead of `clear_records`.